ZeldaFullEditor/Gui/MainTabs/OverworldEditor.cs

```diff
@@ -2427,5 +2547,223 @@ public void SendMapProperties(OverworldMap map)
             NetZS.client.SendMessage(msg, NetDeliveryMethod.ReliableOrdered);
             NetZS.client.FlushSendQueue();
         }
+
+        private void music1Box_SelectedIndexChanged(object sender, EventArgs e)
+        {
+            if (!fromForm)
+            {
+                this.scene.ow.AllMaps[this.scene.selectedMapParent].Music[0] = (byte)((ambient1Box.SelectedIndex << 4) + music1Box.SelectedIndex);
+                this.scene.ow.AllMaps[this.scene.selectedMapParent].Music[1] = (byte)((ambient2Box.SelectedIndex << 4) + music2Box.SelectedIndex);
+                this.scene.ow.AllMaps[this.scene.selectedMapParent].Music[2] = (byte)((ambient3Box.SelectedIndex << 4) + music3Box.SelectedIndex);
+                this.scene.ow.AllMaps[this.scene.selectedMapParent].Music[3] = (byte)((ambient4Box.SelectedIndex << 4) + music4Box.SelectedIndex);
+            }
+        }
+
+        byte[] previewSheets = new byte[1];
+        byte globalPalPreview = 8;
+        bool ispalPreview = false;
+
+        private void OWProperty_SPRGFX_MouseEnter(object sender, EventArgs e)
+        {
+            ispalPreview = false;
+            previewsheetPicturebox.Location = new Point(0, (sender as Hexbox).Location.Y + 48);
+
+            previewSheets = new byte[4];
+            for(int i = 0; i< 4; i++)
+            {
+                previewSheets[i] = (byte)(ROM.DATA[Constants.sprite_blockset_pointer + (OWProperty_SPRGFX.HexValue * 4) + i] + 115);
+            }
+
+            previewsheetPicturebox.Height = (64 * 4);
+            previewsheetPicturebox.Visible = true;
+            previewsheetPicturebox.Refresh();
+        }
+
+        private void OWProperty_SPRGFX_MouseLeave(object sender, EventArgs e)
+        {
+            previewsheetPicturebox.Visible = false;
+        }
+
+        private void previewsheetPicturebox_Paint(object sender, PaintEventArgs e)
+        {
+            if (ispalPreview)
+            {
+                for(int i = 0; i<256;i++)
+                {
+                    e.Graphics.FillRectangle(new SolidBrush(scene.ow.AllMaps[scene.selectedMapParent].GFXBitmap.Palette.Entries[i]), new Rectangle((i%16)*16, (i/16)*16, 16, 16));
+                }
+            }
+            else
+            {
+                for (int i = 0; i < previewSheets.Length; i++)
+                {
+                    if (previewSheets[i] >= 0xDE)
```

Comment:
constant for sheet limit?

Reply:
another one for @Zarby89 i'm not sure what this is for